src/components/PopoverButton/index.tsx

```diff
@@ -0,0 +1,49 @@
+import { useRef, useState } from "react";
+import Button, { ButtonProps } from "@leafygreen-ui/button";
+import Card from "@leafygreen-ui/card";
+import Popover from "@leafygreen-ui/popover";
+import Icon from "components/Icon";
+import { useOnClickOutside } from "hooks";
+
+interface PopoverButtonProps extends Omit<ButtonProps, "children"> {
+  children: React.ReactNode;
+  onClick?: () => void;
+  buttonText: string;
+}
+const PopoverButton: React.FC<PopoverButtonProps> = ({
+  children,
+  onClick,
+  buttonText,
+  ...rest
+}) => {
+  const [isOpen, setIsOpen] = useState(false);
+  const buttonRef = useRef<HTMLButtonElement>(null);
+  const popoverRef = useRef<HTMLDivElement>(null);
+
+  useOnClickOutside([buttonRef, popoverRef], () => {
+    setIsOpen(false);
+  });
+
+  const handleClick = () => {
+    setIsOpen(!isOpen);
+    onClick?.();
+  };
+  return (
+    <Button
+      ref={buttonRef}
+      onClick={handleClick}
+      rightGlyph={<Icon glyph="CaretDown" />}
+      size="small"
+      {...rest}
+    >
+      {buttonText}
+      <Popover active={isOpen} onClick={(e) => e.stopPropagation()}>
+        <div ref={popoverRef}>
+          <Card>{children}</Card>
```

Comment:
[nit] Since this component doesn't specifically mention use of cards, should we maybe omit the Card here and have it included in the child component? Or we could rename it to `PopoverCard` to be clearer.